File: level_1/콜라츠 추측.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int collatz(long long num, int cnt)
{
    printf("%d", cnt);
    if (num == 1)
        return cnt;
    if (cnt == 500)
        return -1;
    num % 2 == 0 ? collatz(num / 2, cnt + 1) : collatz(3 * num + 1, cnt + 1);
}

int solution(int num)
{
    return collatz(num, 0);
}

File: level_1/콜라츠 추측.java
class Solution {
    public int solution(int num) {
        int count = 0;
        return collatz((long)num, count);
    }
    
    public int collatz(long num, int cnt)
    {
        if (num == 1)
            return cnt;
        if (cnt == 500)
            return -1;
        
        return (num % 2 == 0) ? collatz(num / 2, cnt + 1) : collatz(num * 3 + 1, cnt + 1);
    }
}